{"uid":"3c4d354183a0ddd1","name":"test_precision[Test for Kreivo]","fullName":"tests.test_network_parameters.TestPrecision#test_precision","historyId":"e18b324a1a112e7efb6fc4938d7372f6","time":{"start":1716220991562,"stop":1716220991562,"duration":0},"status":"broken","statusMessage":"tests.conftest.FixtureTimeoutError: Timeout 15 when initializing properties: Kreivo","statusTrace":"timeout = 15\nfunction = <bound method Chain.init_properties of <scripts.utils.chain_model.Chain object at 0x7f7d0d7a7400>>\nargs = (), error_message = 'Timeout 15 when initializing properties: Kreivo'\n\n    def execute_with_timeout(timeout: int, function, args: tuple, error_message: str):\n        try:\n>           return func_timeout(timeout=timeout, func=function, args=args)\n\ntests/conftest.py:29: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\ntimeout = 15\nfunc = <bound method Chain.init_properties of <scripts.utils.chain_model.Chain object at 0x7f7d0d7a7400>>\nargs = (), kwargs = {}\n\n    def func_timeout(timeout, func, args=(), kwargs=None):\n        '''\n            func_timeout - Runs the given function for up to #timeout# seconds.\n    \n            Raises any exceptions #func# would raise, returns what #func# would return (unless timeout is exceeded), in which case it raises FunctionTimedOut\n    \n            @param timeout <float> - Maximum number of seconds to run #func# before terminating\n    \n            @param func <function> - The function to call\n    \n            @param args    <tuple> - Any ordered arguments to pass to the function\n    \n            @param kwargs  <dict/None> - Keyword arguments to pass to the function.\n    \n    \n            @raises - FunctionTimedOut if #timeout# is exceeded, otherwise anything #func# could raise will be raised\n    \n            If the timeout is exceeded, FunctionTimedOut will be raised within the context of the called function every two seconds until it terminates,\n            but will not block the calling thread (a new thread will be created to perform the join). If possible, you should try/except FunctionTimedOut\n            to return cleanly, but in most cases it will 'just work'.\n    \n            @return - The return value that #func# gives\n        '''\n    \n        if not kwargs:\n            kwargs = {}\n        if not args:\n            args = ()\n    \n        ret = []\n        exception = []\n        isStopped = False\n    \n        def funcwrap(args2, kwargs2):\n            try:\n                ret.append( func(*args2, **kwargs2) )\n            except FunctionTimedOut:\n                # Don't print traceback to stderr if we time out\n                pass\n            except Exception as e:\n                exc_info = sys.exc_info()\n                if isStopped is False:\n                    # Assemble the alternate traceback, excluding this function\n                    #  from the trace (by going to next frame)\n                    # Pytohn3 reads native from __traceback__,\n                    # python2 has a different form for \"raise\"\n                    e.__traceback__ = exc_info[2].tb_next\n                    exception.append( e )\n    \n        thread = StoppableThread(target=funcwrap, args=(args, kwargs))\n        thread.daemon = True\n    \n        thread.start()\n        thread.join(timeout)\n    \n        stopException = None\n        if thread.is_alive():\n            isStopped = True\n    \n            class FunctionTimedOutTempType(FunctionTimedOut):\n                def __init__(self):\n                    return FunctionTimedOut.__init__(self, '', timeout, func, args, kwargs)\n    \n            FunctionTimedOutTemp = type('FunctionTimedOut' + str( hash( \"%d_%d_%d_%d\" %(id(timeout), id(func), id(args), id(kwargs))) ), FunctionTimedOutTempType.__bases__, dict(FunctionTimedOutTempType.__dict__))\n    \n            stopException = FunctionTimedOutTemp\n            thread._stopThread(stopException)\n            thread.join(min(.1, timeout / 50.0))\n>           raise FunctionTimedOut('', timeout, func, args, kwargs)\nE           func_timeout.exceptions.FunctionTimedOut: Function init_properties (args=()) (kwargs={}) timed out after 15.000000 seconds.\n\n.venv/lib/python3.10/site-packages/func_timeout/dafunc.py:101: FunctionTimedOut\n\nDuring handling of the above exception, another exception occurred:\n\nrequest = <SubRequest 'chain_model' for <Function test_precision[Test for Kreivo]>>\n\n    @pytest.fixture(scope=\"module\", params=get_substrate_chains(), ids=chain_names)\n    def chain_model(request) -> Chain:\n        chain = request.param\n        execute_with_timeout(\n            timeout=FIXTURE_TIMEOUT,\n            function=chain.create_connection,\n            args=(),\n            error_message=f'Timeout {FIXTURE_TIMEOUT} when creating connection: {chain.name}'\n        )\n>       execute_with_timeout(\n            timeout=FIXTURE_TIMEOUT,\n            function=chain.init_properties,\n            args=(),\n            error_message=f'Timeout {FIXTURE_TIMEOUT} when initializing properties: {chain.name}'\n        )\n\ntests/conftest.py:64: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\ntimeout = 15\nfunction = <bound method Chain.init_properties of <scripts.utils.chain_model.Chain object at 0x7f7d0d7a7400>>\nargs = (), error_message = 'Timeout 15 when initializing properties: Kreivo'\n\n    def execute_with_timeout(timeout: int, function, args: tuple, error_message: str):\n        try:\n            return func_timeout(timeout=timeout, func=function, args=args)\n        except FunctionTimedOut:\n            with allure.step(\"Function timed out\"):\n                allure.attach(name=\"Error\", body=error_message,\n                              attachment_type=allure.attachment_type.TEXT)\n>           raise FixtureTimeoutError(error_message)\nE           tests.conftest.FixtureTimeoutError: Timeout 15 when initializing properties: Kreivo\n\ntests/conftest.py:34: FixtureTimeoutError","flaky":false,"newFailed":false,"newBroken":true,"newPassed":false,"retriesCount":2,"retriesStatusChange":false,"beforeStages":[{"name":"chain_model","time":{"start":1716220991562,"stop":1716221007892,"duration":16330},"status":"broken","statusMessage":"tests.conftest.FixtureTimeoutError: Timeout 15 when initializing properties: Kreivo\n","statusTrace":"  File \"/home/runner/work/test-runner/test-runner/nova-utils/.venv/lib/python3.10/site-packages/pluggy/_callers.py\", line 77, in _multicall\n    res = hook_impl.function(*args)\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/.venv/lib/python3.10/site-packages/_pytest/fixtures.py\", line 1128, in pytest_fixture_setup\n    result = call_fixture_func(fixturefunc, request, kwargs)\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/.venv/lib/python3.10/site-packages/_pytest/fixtures.py\", line 907, in call_fixture_func\n    fixture_result = fixturefunc(**kwargs)\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/tests/conftest.py\", line 64, in chain_model\n    execute_with_timeout(\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/tests/conftest.py\", line 34, in execute_with_timeout\n    raise FixtureTimeoutError(error_message)\n","steps":[{"name":"Function timed out","time":{"start":1716221007892,"stop":1716221007892,"duration":0},"status":"passed","steps":[],"attachments":[{"uid":"ac083f241052a1e2","name":"Error","source":"ac083f241052a1e2.txt","type":"text/plain","size":47}],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":1,"hasContent":true,"attachmentStep":false}],"attachments":[],"parameters":[],"shouldDisplayMessage":true,"stepsCount":1,"attachmentsCount":1,"hasContent":true,"attachmentStep":false}],"testStage":{"status":"broken","statusMessage":"tests.conftest.FixtureTimeoutError: Timeout 15 when initializing properties: Kreivo","statusTrace":"timeout = 15\nfunction = <bound method Chain.init_properties of <scripts.utils.chain_model.Chain object at 0x7f7d0d7a7400>>\nargs = (), error_message = 'Timeout 15 when initializing properties: Kreivo'\n\n    def execute_with_timeout(timeout: int, function, args: tuple, error_message: str):\n        try:\n>           return func_timeout(timeout=timeout, func=function, args=args)\n\ntests/conftest.py:29: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\ntimeout = 15\nfunc = <bound method Chain.init_properties of <scripts.utils.chain_model.Chain object at 0x7f7d0d7a7400>>\nargs = (), kwargs = {}\n\n    def func_timeout(timeout, func, args=(), kwargs=None):\n        '''\n            func_timeout - Runs the given function for up to #timeout# seconds.\n    \n            Raises any exceptions #func# would raise, returns what #func# would return (unless timeout is exceeded), in which case it raises FunctionTimedOut\n    \n            @param timeout <float> - Maximum number of seconds to run #func# before terminating\n    \n            @param func <function> - The function to call\n    \n            @param args    <tuple> - Any ordered arguments to pass to the function\n    \n            @param kwargs  <dict/None> - Keyword arguments to pass to the function.\n    \n    \n            @raises - FunctionTimedOut if #timeout# is exceeded, otherwise anything #func# could raise will be raised\n    \n            If the timeout is exceeded, FunctionTimedOut will be raised within the context of the called function every two seconds until it terminates,\n            but will not block the calling thread (a new thread will be created to perform the join). If possible, you should try/except FunctionTimedOut\n            to return cleanly, but in most cases it will 'just work'.\n    \n            @return - The return value that #func# gives\n        '''\n    \n        if not kwargs:\n            kwargs = {}\n        if not args:\n            args = ()\n    \n        ret = []\n        exception = []\n        isStopped = False\n    \n        def funcwrap(args2, kwargs2):\n            try:\n                ret.append( func(*args2, **kwargs2) )\n            except FunctionTimedOut:\n                # Don't print traceback to stderr if we time out\n                pass\n            except Exception as e:\n                exc_info = sys.exc_info()\n                if isStopped is False:\n                    # Assemble the alternate traceback, excluding this function\n                    #  from the trace (by going to next frame)\n                    # Pytohn3 reads native from __traceback__,\n                    # python2 has a different form for \"raise\"\n                    e.__traceback__ = exc_info[2].tb_next\n                    exception.append( e )\n    \n        thread = StoppableThread(target=funcwrap, args=(args, kwargs))\n        thread.daemon = True\n    \n        thread.start()\n        thread.join(timeout)\n    \n        stopException = None\n        if thread.is_alive():\n            isStopped = True\n    \n            class FunctionTimedOutTempType(FunctionTimedOut):\n                def __init__(self):\n                    return FunctionTimedOut.__init__(self, '', timeout, func, args, kwargs)\n    \n            FunctionTimedOutTemp = type('FunctionTimedOut' + str( hash( \"%d_%d_%d_%d\" %(id(timeout), id(func), id(args), id(kwargs))) ), FunctionTimedOutTempType.__bases__, dict(FunctionTimedOutTempType.__dict__))\n    \n            stopException = FunctionTimedOutTemp\n            thread._stopThread(stopException)\n            thread.join(min(.1, timeout / 50.0))\n>           raise FunctionTimedOut('', timeout, func, args, kwargs)\nE           func_timeout.exceptions.FunctionTimedOut: Function init_properties (args=()) (kwargs={}) timed out after 15.000000 seconds.\n\n.venv/lib/python3.10/site-packages/func_timeout/dafunc.py:101: FunctionTimedOut\n\nDuring handling of the above exception, another exception occurred:\n\nrequest = <SubRequest 'chain_model' for <Function test_precision[Test for Kreivo]>>\n\n    @pytest.fixture(scope=\"module\", params=get_substrate_chains(), ids=chain_names)\n    def chain_model(request) -> Chain:\n        chain = request.param\n        execute_with_timeout(\n            timeout=FIXTURE_TIMEOUT,\n            function=chain.create_connection,\n            args=(),\n            error_message=f'Timeout {FIXTURE_TIMEOUT} when creating connection: {chain.name}'\n        )\n>       execute_with_timeout(\n            timeout=FIXTURE_TIMEOUT,\n            function=chain.init_properties,\n            args=(),\n            error_message=f'Timeout {FIXTURE_TIMEOUT} when initializing properties: {chain.name}'\n        )\n\ntests/conftest.py:64: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\ntimeout = 15\nfunction = <bound method Chain.init_properties of <scripts.utils.chain_model.Chain object at 0x7f7d0d7a7400>>\nargs = (), error_message = 'Timeout 15 when initializing properties: Kreivo'\n\n    def execute_with_timeout(timeout: int, function, args: tuple, error_message: str):\n        try:\n            return func_timeout(timeout=timeout, func=function, args=args)\n        except FunctionTimedOut:\n            with allure.step(\"Function timed out\"):\n                allure.attach(name=\"Error\", body=error_message,\n                              attachment_type=allure.attachment_type.TEXT)\n>           raise FixtureTimeoutError(error_message)\nE           tests.conftest.FixtureTimeoutError: Timeout 15 when initializing properties: Kreivo\n\ntests/conftest.py:34: FixtureTimeoutError","steps":[],"attachments":[{"uid":"6080e84467eeaaa1","name":"stdout","source":"6080e84467eeaaa1.txt","type":"text/plain","size":8838}],"parameters":[],"shouldDisplayMessage":true,"stepsCount":0,"attachmentsCount":1,"hasContent":true,"attachmentStep":false},"afterStages":[],"labels":[{"name":"parentSuite","value":"tests"},{"name":"suite","value":"test_network_parameters"},{"name":"subSuite","value":"TestPrecision"},{"name":"host","value":"fv-az888-662"},{"name":"thread","value":"2070-MainThread"},{"name":"framework","value":"pytest"},{"name":"language","value":"cpython3"},{"name":"package","value":"tests.test_network_parameters"},{"name":"resultFormat","value":"allure2"}],"parameters":[{"name":"chain_model","value":"<scripts.utils.chain_model.Chain object at 0x7f7d0d7a7400>"}],"links":[],"hidden":false,"retry":false,"extra":{"severity":"normal","retries":[{"uid":"7896abf363890659","status":"broken","statusDetails":"tests.conftest.FixtureTimeoutError: Timeout 15 when initializing properties: Kreivo","time":{"start":1716220970283,"stop":1716220970283,"duration":0}},{"uid":"45573a9e9e0f15ee","status":"broken","statusDetails":"tests.conftest.FixtureTimeoutError: Timeout 15 when initializing properties: Kreivo","time":{"start":1716220948888,"stop":1716220948888,"duration":0}}],"categories":[{"name":"Test defects","matchedStatuses":[],"flaky":false}],"history":{"statistic":{"failed":0,"broken":3,"skipped":0,"passed":23,"unknown":0,"total":26},"items":[{"uid":"9d9c7f697eaf44b1","reportUrl":"https://novasamatech.github.io/test-runner/5281//#testresult/9d9c7f697eaf44b1","status":"passed","time":{"start":1716192151912,"stop":1716192151912,"duration":0}},{"uid":"ec5893341deedb80","reportUrl":"https://novasamatech.github.io/test-runner/5282//#testresult/ec5893341deedb80","status":"passed","time":{"start":1716206567043,"stop":1716206567043,"duration":0}},{"uid":"362122feffc98197","reportUrl":"https://novasamatech.github.io/test-runner/5280//#testresult/362122feffc98197","status":"passed","time":{"start":1716177747063,"stop":1716177747063,"duration":0}},{"uid":"bf41244ce3e5bcc8","reportUrl":"https://novasamatech.github.io/test-runner/5278//#testresult/bf41244ce3e5bcc8","status":"passed","time":{"start":1716148948839,"stop":1716148948839,"duration":0}},{"uid":"beb27da0ee6bd99b","reportUrl":"https://novasamatech.github.io/test-runner/5279//#testresult/beb27da0ee6bd99b","status":"passed","time":{"start":1716163390876,"stop":1716163390876,"duration":0}},{"uid":"9254c69f91df7f8e","reportUrl":"https://novasamatech.github.io/test-runner/5277//#testresult/9254c69f91df7f8e","status":"broken","statusDetails":"tests.conftest.FixtureTimeoutError: Timeout 15 when creating connection: Kreivo","time":{"start":1716134580892,"stop":1716134580892,"duration":0}},{"uid":"4b2548c539fdef14","reportUrl":"https://novasamatech.github.io/test-runner/5276//#testresult/4b2548c539fdef14","status":"broken","statusDetails":"tests.conftest.FixtureTimeoutError: Timeout 15 when creating connection: Kreivo","time":{"start":1716120202474,"stop":1716120202474,"duration":0}},{"uid":"81bcc3a96665b472","reportUrl":"https://novasamatech.github.io/test-runner/5274//#testresult/81bcc3a96665b472","status":"passed","time":{"start":1716091360430,"stop":1716091360430,"duration":0}},{"uid":"691fe571e1f4d170","reportUrl":"https://novasamatech.github.io/test-runner/5275//#testresult/691fe571e1f4d170","status":"passed","time":{"start":1716105751830,"stop":1716105751830,"duration":0}},{"uid":"2e6736691f46d59a","reportUrl":"https://novasamatech.github.io/test-runner/5272//#testresult/2e6736691f46d59a","status":"passed","time":{"start":1716062552833,"stop":1716062552833,"duration":0}},{"uid":"d7df4c9f35ecc78b","reportUrl":"https://novasamatech.github.io/test-runner/5273//#testresult/d7df4c9f35ecc78b","status":"passed","time":{"start":1716076983309,"stop":1716076983309,"duration":0}},{"uid":"60bca405b4a9c1ae","reportUrl":"https://novasamatech.github.io/test-runner/5271//#testresult/60bca405b4a9c1ae","status":"passed","time":{"start":1716048164586,"stop":1716048164587,"duration":1}},{"uid":"4dc35b2254812eba","reportUrl":"https://novasamatech.github.io/test-runner/5269//#testresult/4dc35b2254812eba","status":"passed","time":{"start":1716019357411,"stop":1716019357411,"duration":0}},{"uid":"cb3d8f9c1e95136e","reportUrl":"https://novasamatech.github.io/test-runner/5270//#testresult/cb3d8f9c1e95136e","status":"passed","time":{"start":1716033773469,"stop":1716033773469,"duration":0}},{"uid":"4f1a718590f444c1","reportUrl":"https://novasamatech.github.io/test-runner/5268//#testresult/4f1a718590f444c1","status":"passed","time":{"start":1716004956179,"stop":1716004956179,"duration":0}},{"uid":"4a5cb05f7c5dde3","reportUrl":"https://novasamatech.github.io/test-runner/5267//#testresult/4a5cb05f7c5dde3","status":"passed","time":{"start":1715990594708,"stop":1715990594708,"duration":0}},{"uid":"abf29211401036fc","reportUrl":"https://novasamatech.github.io/test-runner/5266//#testresult/abf29211401036fc","status":"passed","time":{"start":1715976157491,"stop":1715976157491,"duration":0}},{"uid":"e860b08b9b145379","reportUrl":"https://novasamatech.github.io/test-runner/5264//#testresult/e860b08b9b145379","status":"passed","time":{"start":1715947361035,"stop":1715947361035,"duration":0}},{"uid":"24f9ec2aeac02455","reportUrl":"https://novasamatech.github.io/test-runner/5265//#testresult/24f9ec2aeac02455","status":"passed","time":{"start":1715961761900,"stop":1715961761900,"duration":0}},{"uid":"21425303f5c1df11","reportUrl":"https://novasamatech.github.io/test-runner/5263//#testresult/21425303f5c1df11","status":"passed","time":{"start":1715932957373,"stop":1715932957374,"duration":1}}]},"tags":[]},"source":"3c4d354183a0ddd1.json","parameterValues":["<scripts.utils.chain_model.Chain object at 0x7f7d0d7a7400>"]}